acme/acme/client.py

```diff
@@ -84,6 +84,32 @@ def _send_recv_regr(self, regr, body):
             response, uri=regr.uri,
             terms_of_service=regr.terms_of_service)
 
+    def _post_as_get(self, *args, **kwargs):
+        """
+        Send GET request using the POST-as-GET protocol if needed.
+        The request will be first issued using POST-as-GET for ACME v2. If the ACME CA servers do
+        not support this yet and return an error, request will be retried using GET.
+        For ACME v1, only GET request will be tried, as POST-as-GET is not supported.
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        if self.acme_version >= 2:
+            # We add an empty payload for POST-as-GET requests
+            new_args = args[:1] + (None,) + args[1:]
+            try:
+                return self._post(*new_args, **kwargs)  # pylint: disable=star-args
+            except messages.Error as error:
+                if error.code == 'malformed':
```

Comment:
`malformed` is for sure what you get once post-as-get has been implemented and you send something bad to that endpoint, but I can't find where in the spec it says what to give when you post to a get, pre-post-as-get. Do you have a pointer to that section?

Reply:
There is no explicit declaration in the spec that `malformed` will be returned for requests that uses post-as-get on servers that do not support it.

However, `malformed` is referred to represent errors about invalid request messages in section 5.5. A request that uses an invalid verb would naturally fall in this category. And it is what I observe for Boulder v1 and Boulder v2 without post-as-get support.

So I think it is a reasonable condition to trigger a retry as GET, and ensure retro-compatibility.